### PR TITLE
Added support for empty and no headers sources

### LINKED ISSUE #176: Ability to check table without headers?
(This would obviously not run checks related to headers.)

```
(pilot-dm4t) legend:refit-dirty dan (master)$ head archive/House1.csv 
1381323977,523,74,0,69,0,0,0,0,0,1
1381323990,526,75,0,69,0,0,0,0,0,1
1381324005,540,74,0,68,0,0,0,0,0,1
1381324020,532,74,0,68,0,0,0,0,0,1
1381324035,540,74,0,69,0,0,0,0,0,1
```

Adding a `"dialect": {"header": False}` to resource has no effect.

`datapackage.json`:

```
...
"resources": [
    {
      "name": "house_1.csv",
      "path": "archive/House1.csv",
      "dialect": {
        "header": false
      },
...
```

I would opt to just skip the header-related checks on command line, but can't see how (#175)

---

# Tasks

- [x] support inspection for tables without headers
- [x] support inspection for data package resources without headers (now it's possible with supported CSV dialect `header` option- https://github.com/frictionlessdata/datapackage-py/blob/master/tests/test_resource.py#L517-L528) - MOVED #174

### PR DESCRIPTION
**Pull request by roll**
- fixes #176
- connects https://github.com/frictionlessdata/goodtables.io/issues/225
- fixed `inspect.get_func_args` warning

---

In this PR we just fix places where we had a wrong assumption that `headers` can't be None. It allows:
- inspect tables with `headers=None`
- inspect empty tables (was a reason of GTIO error above)

### CODE REVIEW

**Comment by roll** [comment]
@brew 
Could you please review

File: goodtables/inspector.py [hunk 1]
```diff
@@ -323,7 +324,10 @@ def _prepare_checks(setup, custom, order_fields, infer_fields):
 
     # Bind options
     for check in checks:
-        parameters = inspect.signature(check['func']).parameters
+        if six.PY2:
```
Comment (brew) [inline goodtables/inspector.py]: Perhaps just use the PY2 version and remove the conditional?

Comment (roll) [inline goodtables/inspector.py]: It's deprecated in Python3:
> DeprecationWarning: inspect.getargspec() is deprecated, use inspect.signature() instead

**Comment by roll** [comment]
Thanks Brook!

**Pull request merged**